esti/catalog_export_test.go

```diff
@@ -0,0 +1,418 @@
+	t.Helper()
+	for _, objects := range objectsGroups {
+		for path, obj := range objects {
+			resp, err := uploadContent(ctx, repo, branch, path, obj)
+			require.NoError(t, err)
+			require.Equal(t, http.StatusCreated, resp.StatusCode())
+		}
+	}
+
+	commitResp, err := client.CommitWithResponse(ctx, repo, mainBranch, &apigen.CommitParams{}, apigen.CommitJSONRequestBody{
+		Message: "Table Data",
+	})
+	require.NoErrorf(t, err, "failed commiting uploaded objects to lakefs://%s/%s", repo, branch)
+	require.Equal(t, http.StatusCreated, commitResp.StatusCode())
+	return commitResp.JSON201
+}
+func genCSVData(cols []string, n int) string {
+	csvBlob := ""
+	// generate header
+	for _, c := range cols {
+		csvBlob += c + ","
+	}
+	csvBlob = strings.TrimSuffix(csvBlob, ",") + "\n"
+	// generate rows
+	for rowNum := 0; rowNum < n; rowNum++ {
+		row := ""
+		for i := range cols {
+			row += fmt.Sprintf("%d,", rowNum+i)
+		}
+		csvBlob += strings.TrimSuffix(row, ",") + "\n"
+	}
+	return csvBlob
+}
+
+func testSymlinkS3Exporter(t *testing.T, ctx context.Context, repo string, tablePaths map[string]string, testData *exportHooksTestData) (string, string) {
+	t.Helper()
+
+	tableYaml, err := yaml.Marshal(&testData.TableSpec)
+
+	if err != nil {
+		require.NoError(t, err, "failed marshaling table spec to YAML")
+	}
+
+	hookFiles := map[string]string{
+		testData.SymlinkScriptPath:   symlinkExporterScript,
+		testData.TableDescriptorPath: renderTplAsStr(t, testData.TableSpec, "table", string(tableYaml)),
+	}
+
+	// upload table objects and action script
+	uploadAndCommitObjects(t, ctx, repo, mainBranch, hookFiles, tablePaths)
+
+	// upload action
+	commit := uploadAndCommitObjects(t, ctx, repo, mainBranch, map[string]string{
+		"_lakefs_actions/animals_symlink.yaml": renderTplAsStr(t, testData, "action", symlinkExporterAction),
+	})
+
+	// wait until actions finish running
+	runs := waitForListRepositoryRunsLen(ctx, t, repo, commit.Id, 1)
+	require.Equal(t, "completed", runs.Results[0].Status, "symlink action result not finished")
+
+	// list symlink.txt files from blockstore
+
+	repoResponse, err := client.GetRepositoryWithResponse(ctx, repo)
+	require.NoError(t, err, "could not get repository information")
+	require.Equal(t, repoResponse.StatusCode(), http.StatusOK, "could not get repository information")
+	namespace := repoResponse.JSON200.StorageNamespace
+
+	symlinksPrefix := fmt.Sprintf("%s/_lakefs/exported/%s/%s/animals/", namespace, mainBranch, commit.Id[:6])
+	storageURL, err := url.Parse(symlinksPrefix)
+	require.NoError(t, err, "failed extracting bucket name")
+
+	s3Client, err := testutil.SetupTestS3Client("https://s3.amazonaws.com", testData.AccessKeyId, testData.SecretAccessKey)
+
+	require.NoError(t, err, "failed creating s3 client")
+
+	// we need to retry for the result
+	var symlinkLocations []string
+
+	bo := backoff.NewExponentialBackOff()
+	bo.MaxInterval = 5 * time.Second
+	bo.MaxElapsedTime = 30 * time.Second
```

Comment:
Given that `ExponentialBackoff` is [an exported type](https://pkg.go.dev/github.com/cenkalti/backoff/v4#ExponentialBackOff), why not just
```go
bo := &backoff.ExponentialBackoff{
	MaxInterval:    5 * time.Second,
	MaxElapsedTime: 30 * time.Second,
}
```
?

Reply:
Because it has other variables inside that New func that we want to use as defaults and are not set when calling `backoff.ExponentialBackoff{}`